lib/components/checkbox/test.js

```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import Checkbox from './';
+import renderer from 'react-test-renderer';
+
+it('renders Checkbox correctly when checked', () => {
```

Comment:
just noting that this test doesn't validate that the Checkbox correctly renders but rather that it renders the same as it previously did. for snapshot testing I personally like naming the tests in a way that corresponds to that…

```js
it( `hasn't had its output unintentionally altered`, … );
```

on the other hand I always find intentional unit tests to be valuable as they can identify the behaviors we intend and consider correct

```js
it( 'renders the circle icon when unchecked', () => {
  const checkbox = shallow( <Checkbox checked={false}></Checkbox> );
  expect( checkbox.someWhere( n => n.hasClass( 'icon-circle' ) ) ).toBeTrue();
  expect( checkbox.someWhere( n => n.hasClass( 'icon-checkmark' ) ) ).toBeFalse();
} );

it( 'renders the checkmark icon when checked', () => {
  const checkbox = shallow( <Checkbox checked={false}></Checkbox> );
  expect( checkbox.someWhere( n => n.hasClass( 'icon-circle' ) ) ).toBeFalse();
  expect( checkbox.someWhere( n => n.hasClass( 'icon-checkmark' ) ) ).toBeTrue();
} );
```

Obviously these tests too are deficient but they should stand more resilient than the snapshot tests which will fail with a high false positive rate (for instance, this test will fail when the checkmark icon gets updated even though the behavior of the app is still good).

Personally I'd be happy to see one of those two options taken here: rename the snapshot tests to indicate what they are doing (they are definitely _not_ telling us if the render is "correct"); or test specific behaviors we want to enforce and allow unrelated changes to not fail the tests.

Reply:
I agree. Tests updated.

Reply:
cool - thanks!